runestone/fitb/js/fitb.js

```diff
@@ -32,91 +32,20 @@ FITB.prototype.init = function (opts) {
     this.useRunestoneServices = opts.useRunestoneServices;
     this.origElem = orig;
     this.divid = orig.id;
-    this.questionArray = [];
     this.correct = null;
-    this.feedbackArray = [];
-    /* this.feedbackArray is an array of array of arrays--each outside element is a blank. Each middle element is a different "incorrect" feedback
-    that is tailored for how the question is incorrectly answered. Each inside array contains 2 elements: the regular expression, then text */
-    this.children = [];   // this contains all of the child elements of the entire tag...
-    this.correctAnswerArray = [];   // This array contains the regular expressions of the correct answers
-
-    this.adoptChildren();
-    this.populateCorrectAnswerArray();
-    this.populateQuestionArray();
-
-    this.casei = false;   // Case insensitive--boolean
-    if ($(this.origElem).data("casei") === true) {
-        this.casei = true;
-    }
-    this.populateFeedbackArray();
+    // See comments in fitb.py for the format of ``feedbackArray`` (which is identical in both files).
+    //
+    // Find the script tag containing JSON and parse it. See `SO <https://stackoverflow.com/questions/9320427/best-practice-for-embedding-arbitrary-json-in-the-dom>`_.
+    this.feedbackArray = JSON.parse(this.scriptSelector(this.origElem).html());
+
     this.createFITBElement();
     this.checkServer("fillb");
```

Comment:
I'm creating a checklist/writing up a document on writing a component...

You should check for the existence of a sid key in the opts passed to the constructor.  If sid is populated (as it will be for grading) then checkServer will do the right thing in the grading interface with respect to populating the correct answer.

In fact the right way to do this is to add an init to `RunestoneBase` that takes care of this for everything and then make sure that each component calls  `RunestoneBase.init.apply(this, arguments)`